File: Sockets/Python/socket_server.py
import socket

# Simple echo server
server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server_socket.bind(('localhost', 8080))
server_socket.listen(1)

conn, addr = server_socket.accept()
data = conn.recv(1024)
print("Received:", data.decode())
conn.send(data)

conn.close()
server_socket.close()

File: Sockets/C++/socket_server.cpp
#include <sys/socket.h>
#include <netinet/in.h>
#include <unistd.h>
#include <iostream>
#include <cstring>

int main() {
    int server_fd, new_socket;
    struct sockaddr_in address;
    int addrlen = sizeof(address);
    char buffer[1024] = {0};

    server_fd = socket(AF_INET, SOCK_STREAM, 0);

    address.sin_family = AF_INET;
    address.sin_addr.s_addr = INADDR_ANY;
    address.sin_port = htons(8080);

    bind(server_fd, (struct sockaddr *)&address, sizeof(address));
    listen(server_fd, 3);

    new_socket = accept(server_fd, (struct sockaddr *)&address, (socklen_t*)&addrlen);
    read(new_socket, buffer, 1024);
    std::cout << "Received: " << buffer << std::endl;
    send(new_socket, buffer, strlen(buffer), 0);

    close(new_socket);
    close(server_fd);

    return 0;
}